src/crewai/agent.py

```diff
@@ -117,16 +117,19 @@ class Agent(BaseAgent):
     def post_init_setup(self):
         self.agent_ops_agent_name = self.role
 
-        if hasattr(self.llm, "model_name"):
-            self._setup_llm_callbacks()
+        # Different llms store the model name in different attributes
+        model_name = self.llm.model_name or self.llm.deployment_name
```

Comment:
Won't this break when calling `model_name` with `AttributeError`?

Reply:
Try to pass any other object there, like a `dict` or a random `BaseModel`. I think it's gonna break but I might be wrong.

Reply:
@gvieira I incorporated your suggestions. Now, we run into another issue when someone passes in an LLM as `{}`. 

I walk through the issue in this loom:

https://www.loom.com/share/527fd6b1e25e4c04a1f8a4910c65cde7?sid=f2dc4d87-6414-4dc2-8aed-7a35784301cf

Reply:
Sweet. Best solution for this is to introduce a protocol/interface, so we limit what we accept. For now, I would say what you've done is good enough, as long as we create one of those [low-hanging fruit tickets](https://linear.app/crewai/view/e1827ba2-0757-46ca-89d7-3d26105e8dce).

Reply:
Added!